pkg/scheduler/dependency_graph.go

```diff
@@ -217,28 +236,32 @@ func (sched Scheduler) newScheduledResource(kind, name string, resDefs map[strin
 		return nil, fmt.Errorf("not a proper resource kind: %s. Expected '%s'",
 			kind, strings.Join(resources.Kinds, "', '"))
 	}
-	r, err := sched.newResource(kind, name, resDefs, gc, resourceTemplate)
+	r, existing, err := sched.newResource(kind, name, resDefs, gc, resourceTemplate, silent)
 	if err != nil {
 		return nil, err
 	}
 
-	return newScheduledResourceFor(r, gc), nil
+	return newScheduledResourceFor(r, gc, existing), nil
 }
 
 func (sched Scheduler) newResource(kind, name string, resDefs map[string]client.ResourceDefinition,
-	gc *GraphContext, resourceTemplate interfaces.ResourceTemplate) (interfaces.Resource, error) {
+	gc *GraphContext, resourceTemplate interfaces.ResourceTemplate, silent bool) (interfaces.Resource, bool, error) {
```

Comment:
:+1: for name change for `newResource`, if after refactors it returns factory, instead of resource.
Clearly we have a mess in logic there which `silent` tries to hide.

Still I think that we should reconsider switching to glog instead of built in log package. These logs are completely unnecessary for user, they are debugging utils, and also basing on `silent` value IMO they should be logged on different levels of verbosity.

Now we have them useful when whole graph is computed, but not for subgraphs, where they are silenced - because this is the purpose of `silence`, right?